src/ajax/viewentry.php

```diff
@@ -43,19 +48,26 @@
          array(
                 'ScreeningMOD' => 0, // LOVD+
                 'Transcript_Variant' => 0,
+                'User' => LEVEL_OWNER,
               );
 
-if (isset($aNeededLevel[$_GET['object']])) {
-    $nNeededLevel = $aNeededLevel[$_GET['object']];
+if (isset($aNeededLevel[$sObject])) {
+    $nNeededLevel = $aNeededLevel[$sObject];
 } else {
     $nNeededLevel = LEVEL_ADMIN;
 }
 
 // Call isAuthorized() on the object. NB: isAuthorized() modifies the global
 // $_AUTH for curators, owners and colleagues.
-if ($_GET['object'] == 'Transcript_Variant') {
-    list($nVariantID, $nTranscriptID) = explode(',', $_GET['id']);
+if ($sObject == 'Transcript_Variant') {
+    list($nVariantID, $nTranscriptID) = explode(',', $nID);
     lovd_isAuthorized('variant', $nVariantID);
+} elseif ($sObject == 'User') {
+    lovd_isAuthorized(strtolower($sObject), $nID);
+    // Users viewing their own profile should see a lot more...
+    if ($_AUTH['id'] == $nID && $_AUTH['level'] < LEVEL_CURATOR) {
+        $_AUTH['level'] = LEVEL_CURATOR;
```

Comment:
I don't like this (setting a fake user level to 'trick' the rest of the code). But I don't have an alternative, at least not a simple one, to get the same result.